shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -94,12 +101,8 @@ public FlutterPluginRegistry getPluginRegistry() {
     return pluginRegistry;
   }
 
-  @NonNull
-  public DartExecutor getDartExecutor() {
-    return dartExecutor;
-  }
 
-  // TODO(mattcarroll): what does this callback actually represent?
+  // TODO(mattcarroll): what does this callback actually represent? nothing uses it...
```

Comment:
It looks like the intent was to notify plugins that hot restart had happened, but is not yet wired up.

flutter/flutter#7160 has details.

Reply:
Try exposing this only to plugins.  Then see if customers need this information in an Android execution context.